src/averageAges.js

```diff
@@ -36,9 +44,29 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
-}
+  // const mothersNames = people.map(person => person.mother);
+  // const womenWithChildren = people
+  //   .filter(person => mothersNames.includes(person.name));
+
+  const women = withChildren
+    ? people.filter(person => person.sex === 'f'
+    && womenWithChildren(people, person))
+    : people.filter(person => person.sex === 'f');
+
+  const womenAges = women.map(person => person.died - person.born);
+
+  const womenAverageAge = womenAges
+    .reduce((firstWoman, secondWoman) =>
+      (firstWoman + secondWoman)) / womenAges.length;
+
+  return womenAverageAge;
+};
+
+function womenWithChildren(people, mother) {
+  return people.some(person => person.mother === mother.name);
+};
```

Comment:
It's ok, but there's no need to create separate function here, you can just put `some` inside ternary above